x/cheqd/types/did_validation.go

```diff
@@ -0,0 +1,220 @@
+package types
+
+import (
+	"github.com/cheqd/cheqd-node/x/cheqd/utils"
+	"regexp"
+)
+
+var SplitDIDRegexp, _        = regexp.Compile(`did:([^:]+?)(:([^:]+?))?:([^:]+)$`)
+var DidNamespaceRegexp, _    = regexp.Compile(`^[a-zA-Z0-9]$`)
+// Base58 only allowed (without OolI and 0)
+var UniqueIDRegexp, _        = regexp.Compile(`^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]*$`)
+// That for groups:
+// Example: did:cheqd:testnet:fafdsffq11213343/path-to-s/ome-external-resource?query#key1???#123
+// 1 - [^/?#]* - all the symbols except / and ? and # . This is the DID part                      (did:cheqd:testnet:fafdsffq11213343)
+// 2 - [^?#]*  - all the symbols except ? and #. it means te section started from /, path-abempty (/path-to-s/ome-external-resource)
+// 3 - \?([^#]*) - group for `query` part but with ? symbol 									  (?query)
+// 4 - [^#]*     - group inside query string, match only exact query                              (query)
+// 5 - #([^#]+[\$]?) - group for fragment, starts with #, includes #                              (#key1???)
```

Comment:
In the example fragment is `#key1???#123`, here it's `#key1???`. Where is the truth?